exporters/otlp/otlpmetric/internal/otlpconfig/envconfig.go

```diff
@@ -138,7 +138,7 @@ func (e *EnvOptionsReader) GetOptionsFromEnv() []GenericOption {
 }
 
 func isInsecureEndpoint(endpoint string) bool {
-	return strings.HasPrefix(strings.ToLower(endpoint), "http://")
+	return strings.HasPrefix(strings.ToLower(endpoint), "http://") || strings.HasPrefix(strings.ToLower(endpoint), "unix://")
```

Comment:
Does it make more sense to just invert the logic and check if it is https?

e.g.

```suggestion
	return !strings.HasPrefix(strings.ToLower(endpoint), "https://")
```

Reply:
Sure, I can make the change. It makes some sense though to be "secure by default" and adding exceptions for the insecure mode. Eg. when something is added or a use-case goes unnoticed, in worst case it needs to be fixed in the codebase with additional exception but it will not break the user's security assumptions.

Reply:
I think I'd agree that secure/confidential by default should be the posture and we should only have exceptions to that where the user has clearly indicated a desire for non-secure/confidential operation.

I wonder whether this should be implemented only for gRPC since I'm not sure it makes sense for HTTP.

Reply:
> I wonder whether this should be implemented only for gRPC since I'm not sure it makes sense for HTTP.

@Aneurysm9 I'm actually planning to do a follow-up that would add support for HTTP. It is correct that atm only gRPC works with unix socket endpoints because of the limitations of the HTTP dialer but I think both should work.